### PR TITLE
Refresh LinPhone ecosystem

### PR DESCRIPTION
**Pull request by c0bw3b**
###### Motivation for this change

Main goal is to cleanup the last dependencies on `mbedtls_1_3` which has been EoL for a while now
\+ `mediastreamer-openh264` update
\+ fix many homepages pointing to non-existing pages under the linphone.org domain
\+ add `bcg729` library

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @7c6f434c 

---

Result of `nix-review` 
<details>
  <summary>12 package were build:</summary>

  - bcg729
  - bctoolbox
  - belcard
  - belle-sip
  - belr
  - bzrtp
  - linphone
  - mediastreamer
  - mediastreamer-openh264
  - msilbc
  - ortp
  - trx
</details>

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/linphone/default.nix [hunk 1]
```diff
@@ -27,21 +26,23 @@ stdenv.mkDerivation rec {
...
-  NIX_CFLAGS_COMPILE = " -Wno-error -I${glib.dev}/include/glib-2.0
-    -I${glib.out}/lib/glib-2.0/include -I${gtk2.dev}/include/gtk-2.0/
-    -I${cairo.dev}/include/cairo -I${pango.dev}/include/pango-1.0
-    -I${gtk2}/lib/gtk-2.0/include
-    -DLIBLINPHONE_GIT_VERSION=\"v${version}\"
-    ";
+  NIX_CFLAGS_COMPILE = [
+    "-Wno-error"
+    "-I${glib.dev}/include/glib-2.0"
+    "-I${glib.out}/lib/glib-2.0/include -I${gtk2.dev}/include/gtk-2.0/"
```
Comment (7c6f434c) [inline pkgs/applications/networking/instant-messengers/linphone/default.nix]: Maybe if it is split into strings, there should be a string per logical part?

Comment (c0bw3b) [reply pkgs/applications/networking/instant-messengers/linphone/default.nix]: Indeed, and that was I intended to do in the first place... Thanks, will fix.

File: pkgs/development/libraries/bctoolbox/default.nix [hunk 2]
```diff
@@ -1,19 +1,23 @@
-{stdenv, fetchFromGitHub, cmake, mbedtls, bcunit, srtp}:
```
Comment (7c6f434c) [inline pkgs/development/libraries/bctoolbox/default.nix]: `srtp` is now broken, and `bcg729` replaces it?

Comment (c0bw3b) [reply pkgs/development/libraries/bctoolbox/default.nix]: No, `srtp` just wasn't used by bctoolbox. It's an an input for mediastreamer2 though.

**Pull request merged**